awx/main/tasks.py

```diff
@@ -660,6 +666,42 @@ def update_host_smart_inventory_memberships():
         smart_inventory.update_computed_fields()
 
 
+@task()
+def migrate_legacy_event_data(tblname):
+    if 'event' not in tblname:
+        return
+    with advisory_lock(f'bigint_migration_{tblname}', wait=False) as acquired:
+        if acquired is False:
+            return
+        chunk = 10000
+
+        def _remaining():
+            try:
+                cursor.execute(f'SELECT MAX(id) FROM _old_{tblname};')
+                return cursor.fetchone()[0]
```

Comment:
This will return `None` if there are no rows (the table exists, but is empty).

Reply:
Maybe `SELECT COUNT(*) FROM pg_tables WHERE tablename LIKE '_old_%'` and look for nonzero

Reply:
Well, I also actually *need* the count.  So looking at `pg_tables` would be _two_ queries.  One to tell if the table exists, and (if it does) one to actually get the `MAX(id)`.

Reply:
Ah ok, didn't realize you were using the count

Reply:
Do we not remove the temporary tables at the end of the process?  They just hang around forever with no rows in them?

Reply:
We do: 

https://github.com/ansible/awx/pull/6032/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R701

Reply:
So if the row count is zero, you still want to re-launch the task, so it can delete the temporary tables ... right?

Reply:
I see what you're getting at, and I just cleaned up the logic a bit so I think it won't require another task to cleanup the table (if the server restarts).